From 00d14068a4bdf6c5a3824dc58faa992807636413 Mon Sep 17 00:00:00 2001
From: Nick Xie <nick@khadas.com>
Date: Mon, 6 Jan 2020 16:52:46 +0800
Subject: [PATCH 11/53] reserve memory for BL32

!!! hyphop - note !!! this patch broke normal uboot behaviour
for $loadaddr usage for many commands
becouse not possible to write into $loadaddr - its reserved

kedge# dhcp
Speed: 1000, full duplex
BOOTP broadcast 1
BOOTP broadcast 2
BOOTP broadcast 3
BOOTP broadcast 4
DHCP client bound to address 172.23.10.200 (3014 ms)
Using ethernet@fe300000 device
TFTP from server 172.23.10.1; our IP address is 172.23.10.200
Filename 'pxelinux.cfg'.

TFTP error: trying to overwrite reserved memory...

Signed-off-by: Nick Xie <nick@khadas.com>
---
 arch/arm/mach-rockchip/sdram.c  |  8 ++++++--
 include/configs/rk3399_common.h | 10 +++++-----
 2 files changed, 11 insertions(+), 7 deletions(-)

diff --git a/arch/arm/mach-rockchip/sdram.c b/arch/arm/mach-rockchip/sdram.c
index 530644c0..06442627 100644
--- a/arch/arm/mach-rockchip/sdram.c
+++ b/arch/arm/mach-rockchip/sdram.c
@@ -39,8 +39,12 @@ int dram_init_banksize(void)
 			 gd->ram_top);
 
 #ifdef CONFIG_ARM64
-	/* Reserve 0x200000 for ATF bl31 */
-	gd->bd->bi_dram[0].start = 0x200000;
+	struct tos_parameter_t *tos_parameter;
+
+	tos_parameter = (struct tos_parameter_t *)(CONFIG_SYS_SDRAM_BASE +
+				TRUST_PARAMETER_OFFSET);
+
+	gd->bd->bi_dram[0].start = tos_parameter->tee_mem.phy_addr + tos_parameter->tee_mem.size;
 	gd->bd->bi_dram[0].size = top - gd->bd->bi_dram[0].start;
 #else
 #ifdef CONFIG_SPL_OPTEE
diff --git a/include/configs/rk3399_common.h b/include/configs/rk3399_common.h
index 89a8a44b..96269514 100644
--- a/include/configs/rk3399_common.h
+++ b/include/configs/rk3399_common.h
@@ -47,11 +47,11 @@
 #ifndef CONFIG_SPL_BUILD
 
 #define ENV_MEM_LAYOUT_SETTINGS \
-	"scriptaddr=0x00500000\0" \
-	"pxefile_addr_r=0x00600000\0" \
-	"fdt_addr_r=0x01f00000\0" \
-	"kernel_addr_r=0x02080000\0" \
-	"ramdisk_addr_r=0x06000000\0"
+	"scriptaddr=0x0a500000\0" \
+	"pxefile_addr_r=0x0a600000\0" \
+	"fdt_addr_r=0x0bf00000\0" \
+	"kernel_addr_r=0x0c080000\0" \
+	"ramdisk_addr_r=0x10000000\0" \
 
 #ifndef ROCKCHIP_DEVICE_SETTINGS
 #define ROCKCHIP_DEVICE_SETTINGS
-- 
2.17.1

